src/lib.rs

```diff
@@ -370,48 +606,123 @@ impl TestHarness {
     }
 
     fn do_create_block(&mut self, tx_hashes: &[crypto::Hash]) {
-        let validator_id = self.state().validator_id().expect(
-            "Tested node is not a validator",
-        );
-        let height = self.state().height();
-        let last_hash = *self.state().last_hash();
-        let round = Round::first();
+        let height = self.height().next();
+        let last_hash = self.last_block_hash();
 
-        let handler = &mut self.handler;
-        let (block_hash, patch) = handler.create_block(validator_id, height, tx_hashes);
-        handler.state.add_block(
-            block_hash,
-            patch,
-            tx_hashes.to_vec(),
-            validator_id,
-        );
+        self.update_configuration();
 
-        let propose = self.network.us().create_propose(
-            height,
-            &last_hash,
-            tx_hashes,
-        );
-        let precommits: Vec<_> = self.network
+        let (block_hash, patch) = {
+            let validator_id = self.leader().validator_id().expect(
+                "Tested node is not a validator",
+            );
+            let transactions = self.mempool();
+            self.blockchain.create_patch(
+                validator_id,
+                height,
+                tx_hashes,
+                &transactions,
+            )
+        };
+
+        // Remove txs from mempool
+        {
+            let mut transactions = self.mempool.write().expect(
+                "Cannot modify transactions in mempool",
+            );
+            for hash in tx_hashes {
+                transactions.remove(hash);
+            }
+        }
+
+        let propose = self.leader().create_propose(height, &last_hash, tx_hashes);
+        let precommits: Vec<_> = self.network()
             .validators()
             .iter()
             .map(|v| v.create_precommit(&propose, &block_hash))
             .collect();
-        handler.commit(block_hash, precommits.iter(), Some(round));
+
+        self.blockchain
+            .commit(&patch, block_hash, precommits.iter())
+            .unwrap();
+
+        self.poll_events();
```

Comment:
What purpose does `poll_events()` serve here?

Reply:
Because code below invokes `handle_commit` what can entail to the occurrence of events with transactions.
```rust
        self.blockchain
            .commit(&patch, block_hash, precommits.iter())
            .unwrap();
```